### PR TITLE
Add information about function names and return types

### PR DESCRIPTION
**Pull request by fdiaz**
**This PR is easier to review commit-by-commit**

This PR adds information about a function name and its return type to:
- Free functions in a file
- Extensions
- Protocols
- Enums
- Classes
- Structs

We should likely add information about the argument labels and parameters, but we can do that in another PR.

### CODE REVIEW

File: Sources/SwiftInspectorVisitors/Tests/FunctionDeclarationVisitorSpec.swift [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+        it("finds the function name") {
+          expect(self.sut.functionDeclarations.first?.name) == "greet"
+        }
+        it("finds the return type") {
+          expect(self.sut.functionDeclarations.first?.returnType) == TypeDescription.simple(name: "String")
+        }
+      }
+
+      context("a function with a return value") {
```
Comment (dfed) [inline Sources/SwiftInspectorVisitors/Tests/FunctionDeclarationVisitorSpec.swift:36]: this function does not have a return value

File: Sources/SwiftInspectorVisitors/Tests/FunctionDeclarationVisitorSpec.swift [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            }
+            """
+
+          try? self.sut.walkContent(content)
+        }
+
+        it("finds the function name") {
+          expect(self.sut.functionDeclarations.first?.name) == "minMax"
+        }
+        it("sets the return type as nil") {
```
Comment (dfed) [inline Sources/SwiftInspectorVisitors/Tests/FunctionDeclarationVisitorSpec.swift:78]: the return type here is not nil

File: Sources/SwiftInspectorVisitors/Tests/FunctionDeclarationVisitorSpec.swift [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+            func printWithoutCounting(string: String) {
+                let _ = printAndCount(string: string)
+            }
+            """
+
+          try? self.sut.walkContent(content)
+        }
+
+        it("finds the correct amount of functions") {
```
Comment (dfed) [inline Sources/SwiftInspectorVisitors/Tests/FunctionDeclarationVisitorSpec.swift:108]: nit: "correct number" not "correct amount"

File: Sources/SwiftInspectorVisitors/Tests/ExtensionVisitorSpec.swift [hunk 4]
```diff
@@ -160,7 +160,10 @@ final class ExtensionVisitorSpec: QuickSpec {
               }
 
               typealias TestClassTypeAlias = TestClass
-            }
+
+              func some() -> Int {
+                1
+              }
```
Comment (dfed) [inline Sources/SwiftInspectorVisitors/Tests/ExtensionVisitorSpec.swift:166]: missing closing parenthesis on the type

File: Sources/SwiftInspectorVisitors/FunctionDeclarationVisitor.swift [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    return .skipChildren
+  }
+
+  fileprivate var returnType: TypeDescription?
+}
+
+public struct FunctionDeclarationInfo: Codable, Hashable {
+  public let name: String
+  public let returnType: TypeDescription?
+}
```
Comment (dfed) [inline Sources/SwiftInspectorVisitors/FunctionDeclarationVisitor.swift:28]: maybe add a TODO here to add parameter info?

**Review by dfed — APPROVED** [review]
LGTM with some nits. Excited to have this data!

File: Sources/SwiftInspectorVisitors/Tests/FunctionDeclarationVisitorSpec.swift (hunk 1 above)
Comment (fdiaz) [reply Sources/SwiftInspectorVisitors/Tests/FunctionDeclarationVisitorSpec.swift:36]: copy-paste 😁

File: Sources/SwiftInspectorVisitors/Tests/FunctionDeclarationVisitorSpec.swift (hunk 2 above)
Comment (fdiaz) [reply Sources/SwiftInspectorVisitors/Tests/FunctionDeclarationVisitorSpec.swift:78]: copy-paste 😁

File: Sources/SwiftInspectorVisitors/NestableTypeVisitor.swift [hunk 6]
```diff
@@ -159,6 +159,10 @@ public final class NestableTypeVisitor: SyntaxVisitor {
         genericRequirementVisitor.walk(genericWhereClause)
       }
 
+      let visitor = FunctionDeclarationVisitor()
```
Comment (bachand) [inline Sources/SwiftInspectorVisitors/NestableTypeVisitor.swift:162]: 👍

**Review by bachand — COMMENTED** [review]
Will defer to @dfed 's review. Very exciting!

**Pull request merged**